### ISSUE TITLE
linter: array-callback-return does not consider throw

### ISSUE THREAD

**Issue by simonbuchan** [oxc-project/oxc#12176]
### What version of Oxlint are you using?

1.6.0

### What command did you run?

oxlint --deny=eslint/array-callback-return test.js

### What does your `.oxlintrc.json` config file look like?

No config for repro

### What happened?

```
  × eslint(array-callback-return): Missing return on some path for array method "Array.prototype.map"
   ╭─[test.js:1:19]
 1 │ ╭─▶ [1, 2].map((x) => {
 2 │ │     if (x % 2) {
 3 │ │       return x * 2;
 4 │ │     } else {
 5 │ │       throw new Error("x is even");
 6 │ │     }
 7 │ ╰─▶ });
 8 │
   ╰────
  help: Array method "Array.prototype.map" needs to have valid return on all code paths

Found 0 warnings and 1 error.
Finished in 4ms on 1 file with 88 rules using 16 threads.
```

Input:

```js
[1, 2].map((x) => {
  if (x % 2) {
    return x * 2;
  } else {
    throw new Error("x is even");
  }
});

[1, 2].map((x) => {
  if (x % 2) {
    return x * 2;
  } else {
    return x + 1;
  }
});
```

ESLint's version of the rule doesn't report on either of the above.

**Comment by mrpinkcat** [comment]
I'm also facing this issue, any news on it ?

**Comment by camc314** [comment]
I think this is fixed via https://github.com/oxc-project/oxc/commit/fd14b39fa9ca5294b29f58b3383b58a7c96e0e4b ?

**Comment by camc314** [comment]
@mrpinkcat @simonbuchan do you mind trying the latest oxlint, and seeing if this is fixed? If it's not can you leave a comment and i'll re-open
